### DISCUSSION TITLE
Google OAuth2 Implementation not storing profile image

### DISCUSSION THREAD

**Discussion by djt** [pocketbase/pocketbase#2797] [Q&A] [answered]
First of all, thanks for providing such an amazing product.  I've only just started playing around with pocketbase and am finding it a joy to work with.

I thought I'd lodge this issue as I had an unexpected result with the Google OAuth2 implementation, where it was not storing my profile image.  I have attached the result from ```await pb.collection('users').authWithOAuth2({ provider: 'google' })```

```
{
    "meta": {
        "id": "115379048999335771247",
        "name": "Daniel",
        "username": "",
        "email": "",
        "avatarUrl": "https://lh3.googleusercontent.com/a/AAcHTtdaksxpJZ3tCJZqdmDXnk-yLCY55REhzCn4XSsx=s96-c",
        "rawUser": {
            "given_name": "Daniel",
            "id": "115379048999335771247",
            "locale": "en",
            "name": "Daniel",
            "picture": "https://lh3.googleusercontent.com/a/AAcHTtdaksxpJZ3tCJZqdmDXnk-yLCY55REhzCn4XSsx=s96-c"
        },
        "accessToken": "***",
        "refreshToken": "",
        "isNew": true
    },
    "record": {
        "avatar": "",
        "collectionId": "_pb_users_auth_",
        "collectionName": "users",
        "created": "2023-06-26 13:29:54.712Z",
        "email": "",
        "emailVisibility": false,
        "id": "2wfzb8xwtf2age1",
        "name": "",
        "updated": "2023-06-26 13:29:54.712Z",
        "username": "users11536",
        "verified": true,
        "expand": {}
    },
    "token": "***"
}
```

Any help would be appreciated - and thanks again!

**Comment by ganigeorgiev** [comment, marked as answer]
The only record fields we autoset with the OAuth2 record creation are the `email` (if available), and the `username` (if available and satisfy the PocketBase username format, otherwise - autogenerate).

Any other record fields need to be updated in a separate `update()` call because they are dynamic and depends on your auth collection configuration.

**Reply by djt** [reply]
Great, thanks for the clarification
